File: app/src/main/java/com/shindra/LoadingDialog.java
package com.shindra.Misc;

import android.app.Activity;
import android.view.LayoutInflater;
import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.Fragment;

import com.shindra.R;

public class LoadingDialog {
    private AlertDialog dialog;

    public LoadingDialog(Activity activity) {
        AlertDialog.Builder builder = new AlertDialog.Builder(activity);
        LayoutInflater inflater = activity.getLayoutInflater();
        builder.setView(inflater.inflate(R.layout.layout_chargement, null));
        builder.setCancelable(false);
        dialog = builder.create();
    }

    public void show() {
        dialog.show();
    }

    public void dismiss() {
        dialog.dismiss();
    }
}

File: app/src/main/java/com/shindra/LoadingDialog.kt
package com.shindra

import android.app.Activity
import android.app.AlertDialog
import android.view.View
import androidx.appcompat.app.AppCompatActivity

class LoadingDialog(context : Activity) : AppCompatActivity() {

    private val loadingDialogView = View.inflate(context, R.layout.loading_dialog,null)
    private val dialogBuilder = AlertDialog.Builder(context).setView(loadingDialogView)
    private val loadingDialog = dialogBuilder.create()

    fun showLoadingDialog() {
        loadingDialog.show()
    }

    fun dismissLoadingDialog() {
        loadingDialog.dismiss()
    }

}